### DISCUSSION TITLE
MDX component overrides

### DISCUSSION THREAD

**Discussion by synecdokey** [withastro/starlight#1107] [Feature Requests]
### What version of `starlight` are you using?

0.12.1

### What is your idea?

Enabling global overrides for MDX components, rather than just the starlight ones.

### Why is this feature necessary?

Anyone with a need to replace the default handling of MDX components systematically would benefit.

In my specific case, I want to be able to provide a way to create interactive code snippets, but importing the component in each file when the code block element could do the trick is somewhat repetitive, and doesn't really show up nicely on GitHub/gitlab previews (Since you are using a `<Component />` rather than a set of backticks). Or it could be a simple copy button on the same code block.

Not having a way to support those in a declarative and systematic manner makes it hard to do these sweeping changes in a way that fits an existing workflow.

### Do you have examples of this feature in other projects?

- https://chakra-ui.com/docs/components/accordion/usage#usage (https://github.com/chakra-ui/chakra-ui-docs/tree/main/src/components/mdx-components)

### Participation

- [x] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thank you for sharing the use case! Noting an overlap with this issue: https://github.com/delucis/astro-auto-import/issues/30

Both would allow ways to automatically add component overrides to your MDX without using the `<Content components={{...components, h1: Heading }} />` [custom components technique](https://docs.astro.build/en/guides/markdown-content/#custom-components-with-imported-mdx) (which Starlight makes difficult to do because we don’t expose `<Content />` rendering).

Implementing in `astro-auto-import` might be a nice option for a generic tool that could do this in Starlight and beyond.

**Reply by synecdokey** [reply]
That sounds definitely better than limiting this to starlight!
